src/dataset.jl

```diff
@@ -178,7 +178,71 @@ function write(
     filename::AbstractString;
     kwargs...,
 )::Nothing
-    destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    drivername = GDAL.gdalgetdrivershortname(getdriver(dataset).ptr)
+    if drivername in gdal_raster_drivers
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif drivername in gdal_vector_drivers
+        writevectords(dataset, filename; kwargs...)
+    else
+        error("Dataset format not recognized.")
+    end
```

Comment:
1. I think I'd almost prefer to have `writerasters(...)` and `writelayers(...)` and deprecate the `write(...)` function if there isn't a consistent GDAL API for it to be writing to both formats.
2. Would an approach of counting number of rasters and layers not work for identifying whether it should write to raster or vector?

Reply:
1. I am not a master of GDAL. But with regard to python even having separate libraries for vector and raster data, it might be justified to split it into separate functions. But one could keep the write function around for backward compatibility or so.
2. That's probably a better idea than looking up the name in an ambiguous list. I'll look into that.